src/charm.py

```diff
@@ -175,7 +175,10 @@ def _mongo_scrape_config(self) -> List[Dict]:
                         "targets": [
                             f"{self.unit_host(self.unit)}:{Config.Monitoring.MONGODB_EXPORTER_PORT}"
                         ],
-                        "labels": {"cluster": self.app.name, "replication_set": self.app.name},
+                        "labels": {
+                            "cluster": self.get_config_server_name() or self.app.name,
+                            "replication_set": self.app.name,
```

Comment:
As discussed - let's proceed step by step on this one and only put it back when we confirm juju app / unit labels work as expected.